src/state_machine.zig

```diff
@@ -594,6 +745,653 @@ test "linked accounts" {
+
+    for (commits) |itm, i| {
+        const create_result = state_machine.commit_transfer(itm.input);
+        switch (itm.outcome) {
+            .ok => try testing.expectEqual(itm.input, state_machine.get_commit(itm.input.id).?.*),
+            else => try testing.expectEqual(itm.outcome, create_result),
+        }
+    }
+
+    // Test Balance AFTER commit
+    // Account 1
+    const acc_1_after = state_machine.get_account(1).?.*;
+    try testing.expectEqual(@as(u64, 30), acc_1_after.debits_accepted); //+15 (acceptance applied)
+    try testing.expectEqual(@as(u64, 45), acc_1_after.debits_reserved); //-15 (reserved moved)
+    try testing.expectEqual(@as(u64, 0), acc_1_after.credits_accepted);
+    try testing.expectEqual(@as(u64, 0), acc_1_after.credits_reserved);
+    // Account 2
+    const acc_2_after = state_machine.get_account(2).?.*;
+    try testing.expectEqual(@as(u64, 0), acc_2_after.debits_accepted);
+    try testing.expectEqual(@as(u64, 0), acc_2_after.debits_reserved);
+    try testing.expectEqual(@as(u64, 30), acc_2_after.credits_accepted); //+15 (acceptance applied)
+    try testing.expectEqual(@as(u64, 45), acc_2_after.credits_reserved); //-15 (reserved moved)
+
+    // Test COMMIT with invalid debit/credit accounts
+    state_machine.create_account_rollback(accounts[3]);
+    try testing.expect(state_machine.get_account(accounts[3].id) == null);
+    try testing.expectEqual(
+        state_machine.commit_transfer(std.mem.zeroInit(Commit, .{
+            .id = 7,
+            .timestamp = (timestamp_cont + 2),
+        })),
+        .credit_account_not_found,
+    );
+    state_machine.create_account_rollback(accounts[2]);
+    try testing.expect(state_machine.get_account(accounts[2].id) == null);
+    try testing.expectEqual(
+        state_machine.commit_transfer(std.mem.zeroInit(Commit, .{
+            .id = 7,
+            .timestamp = (timestamp_cont + 2),
+        })),
+        .debit_account_not_found,
+    );
+
+    state_machine.commit_transfer_rollback(commits[4].input); //rollback [id=2] not rejected
+
+    // Account 1
+    const acc_1_rollback = state_machine.get_account(1).?.*;
+    try testing.expectEqual(@as(u64, 15), acc_1_rollback.debits_accepted); //-15 (rollback)
+    try testing.expectEqual(@as(u64, 60), acc_1_rollback.debits_reserved);
+    try testing.expectEqual(@as(u64, 0), acc_1_rollback.credits_accepted);
+    try testing.expectEqual(@as(u64, 0), acc_1_rollback.credits_reserved);
+    // Account 2
+    const acc_2_rollback = state_machine.get_account(2).?.*;
+    try testing.expectEqual(@as(u64, 0), acc_2_rollback.debits_accepted);
+    try testing.expectEqual(@as(u64, 0), acc_2_rollback.debits_reserved);
+    try testing.expectEqual(@as(u64, 15), acc_2_rollback.credits_accepted); //-15 (rollback)
+    try testing.expectEqual(@as(u64, 60), acc_2_rollback.credits_reserved);
+
+    state_machine.commit_transfer_rollback(commits[7].input); //rollback [id=3] rejected
```

Comment:
These trailing comments perhaps should rather be on the preceding line like how we do the others?